src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -57,8 +60,47 @@ public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, stri
 
             _frameworkIdentifier = frameworkIdentifier;
             _frameworkVersion = NormalizeVersion(frameworkVersion);
-            _frameworkProfile = frameworkProfile ?? string.Empty;
+
+            IsNet5Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = profile ?? string.Empty;
+            Platform = string.Empty;
+            PlatformVersion = FrameworkConstants.EmptyVersion;
+        }
+
+        /// <summary>
+        /// Creates a new NuGetFramework instance, with an optional platform and platformVersion (only available for net5.0+)
+        /// </summary>
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string platform, Version platformVersion)
+        {
+            if (frameworkIdentifier == null)
+            {
+                throw new ArgumentNullException("frameworkIdentifier");
+            }
+
+            if (frameworkVersion == null)
+            {
+                throw new ArgumentNullException("frameworkVersion");
```

Comment:
similar nit/style feedback applies to the rest.

Reply:
see above comment -- I don't think we can use this here.